### PR TITLE
Small tidy up of special variable declarations.

### PR DESCRIPTION
**Pull request by aardvark179**
Smaller diff than it looks, removing the module qualifiers and the indentation change make it look much bigger.

### CODE REVIEW

File: src/main/ruby/core/thread.rb [hunk 1]
```diff
@@ -447,12 +447,7 @@ def inspect
              Truffle::ThreadOperations.set_thread_local(:$SAFE, value) }
 )
 
-Truffle::KernelOperations.define_hooked_variable(
-  :$!,
-  -> { Truffle::ThreadOperations.get_thread_local(:$!) },
-  -> value { Truffle::ThreadOperations.set_thread_local(:$!, value) }
-)
-
+Truffle::KernelOperations.define_read_only_global(:$!, -> { Truffle::ThreadOperations.get_thread_local(:$!) })
```
Comment (eregon) [inline src/main/ruby/core/thread.rb]: So this is actually a bug fix, right?
Could you mention it in the commit message?

File: src/main/ruby/core/truffle/kernel_operations.rb [hunk 2]
```diff
@@ -22,82 +22,83 @@ def self.define_read_only_global(name, getter)
...
+             global_variable_set(:$stdout, v) })
+
+    alias $> $stdout
+
+    define_hooked_variable(
+      :$stderr,
+      -> { global_variable_get(:$stderr) },
+      -> v { raise TypeError, "$stderr must have a write method #{v.class} given." unless v.respond_to?(:write)
+             global_variable_set(:$stderr, v) })
+  end
```
Comment (eregon) [inline src/main/ruby/core/truffle/kernel_operations.rb]: You reordered them but it's same set of globals defined here, right? (The diff makes it hard to see)

File: src/main/ruby/core/truffle/kernel_operations.rb [hunk 3]
```diff
@@ -22,82 +22,83 @@ def self.define_read_only_global(name, getter)
...
 
-Truffle::KernelOperations.define_hooked_variable(
-  :$VERBOSE,
-  -> { Truffle::KernelOperations.global_variable_get(:$VERBOSE) },
-  -> v { v = if v.nil?
-               nil
-             else
-               !!v
+    define_hooked_variable(
+      :$,,
```
Comment (pitr-ch) [inline src/main/ruby/core/truffle/kernel_operations.rb]: This looks really weird :) Could you use `:'$.'`?

File: src/main/ruby/core/truffle/kernel_operations.rb (hunk 2 above)
Comment (aardvark179) [reply src/main/ruby/core/truffle/kernel_operations.rb]: Yes, it's the same set of globals.

File: src/main/ruby/core/thread.rb (hunk 1 above)
Comment (aardvark179) [reply src/main/ruby/core/thread.rb]: Done.

**Pull request merged**